pkg/sshd/lock.go

```diff
@@ -38,9 +38,9 @@ func wrapInLock(lck RepositoryLock, repoName string, fn func() error) error {
 		case <-doneCh:
 		}
 	}()
+	defer lck.Unlock(repoName)
```

Comment:
@kmala this is definitely a good change, but I wasn't previously failing to `git push` a second time